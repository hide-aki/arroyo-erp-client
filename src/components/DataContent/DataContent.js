import React, {memo} from 'react';
import PropTypes from 'prop-types';

const DataContent = memo(({title, children, onClick, style}) =>
  <div style={{...style}} className="data_row" onClick={onClick && onClick}>
      <span className="data_title">
        <b>{title}: </b>
      </span>
    <span className="data_content">{children}</span>
  </div>,
);

DataContent.propTypes = {
  /** The key value passed  */
  title: PropTypes.string.isRequired,
  /** An onClick event func */
  onClick: PropTypes.func,
  /** Custom style properties */
  style: PropTypes.object,
  /** Children to render */
  children: PropTypes.node,
};

export default DataContent;
