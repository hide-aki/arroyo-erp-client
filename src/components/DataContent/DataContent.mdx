---
name: DataContent
menu: Components
route: /components/DataContent
---

import { Props, Playground } from 'docz';
import DataContent from './DataContent';

# DataContent

This component is meant to wrap a simple pair of key:values formatted.

<Props of={DataContent} />

<Playground>
  <DataContent title="Title1" onClick={function() {}} style={{margin: '1rem 0'}}>
    <span>value 1</span>
  </DataContent>
  <DataContent title="Title2" onClick={function() {}} style={{margin: '1rem 0'}}>
    <span>value 2</span>
  </DataContent>
  <DataContent title="Title3" onClick={function() {}} style={{margin: '1rem 0'}}>
    <span>value 3</span>
  </DataContent>
</Playground>

