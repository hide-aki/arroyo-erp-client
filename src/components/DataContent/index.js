import DataContent from './DataContent';

export default DataContent;
