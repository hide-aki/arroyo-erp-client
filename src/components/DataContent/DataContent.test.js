// Container.test.js

import React from 'react';
import Enzyme, {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import DataContent from './DataContent';

Enzyme.configure({adapter: new Adapter()});

describe('<DataContent />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<DataContent title="some title">Some Content</DataContent>);
  });

  it('Renders properly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('It has some title prop', () => {
    expect(wrapper.find({title: 'some title'}));
  });
  it('It renders some content', () => {
    expect(wrapper.find('Some Content'));
  });
});
