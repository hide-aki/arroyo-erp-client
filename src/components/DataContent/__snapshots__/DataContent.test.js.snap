// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataContent /> Renders properly 1`] = `
<Memo()
  title="some title"
>
  <div
    className="data_row"
    style={Object {}}
  >
    <span
      className="data_title"
    >
      <b>
        some title
        : 
      </b>
    </span>
    <span
      className="data_content"
    >
      Some Content
    </span>
  </div>
</Memo()>
`;
